rootfs/api/models/config.py

```diff
@@ -105,6 +111,40 @@ def set_tags(self, previous_config):
 
         raise DeisException(message)
 
+    def set_healthcheck(self, previous_config):
+        data = getattr(previous_config, 'healthcheck', {}).copy()
+        new_data = getattr(self, 'healthcheck', {}).copy()
+        # update the config data for healthcheck if they are not
+        # present for per proctype
+        if 'livenessProbe' in data.keys() or 'readinessProbe' in data.keys():
```

Comment:
Ah that is what you mean by serializer validation, can't even object create things direclty anymore - Please add some comments to that effect and specifically that this can be removed in the next major version, since we've been adding `# TODO` like comments around the code base